python/res/fm/ecl/ecl_run.py

```diff
@@ -164,37 +165,34 @@ def initMPI(self):
                 fileH.write("%s\n" % host)
 
 
-
-    def initRedirect(self):
+    def execSerialEclipse(self):
         stdout_file = "%s.LOG" % self.base_name
         stderr_file = "%s.stderr" % self.base_name
-
-        # Redirecting stdout / stderr
-        fd_stdout = os.open(stdout_file , os.O_WRONLY | os.O_TRUNC | os.O_CREAT , 0o644);
-        fd_stderr = os.open(stderr_file , os.O_WRONLY | os.O_TRUNC | os.O_CREAT , 0o644);
-
-        os.dup2(fd_stdout , 1)
-        os.dup2(fd_stderr , 2)
-
-        os.close(fd_stdout)
-        os.close(fd_stderr)
-
-
-
-    def execSerialEclipse(self):
-        os.execv( self.sim.executable , [ self.sim.executable , self.base_name ])
+        with open(stdout_file, "w") as stdout, open(stderr_file, "w") as stderr:
+            process = subprocess.Popen([self.sim.executable, self.base_name], env=self.sim.env, stdout=stdout, stderr=stderr)
```

Comment:
Please split this line. Something like:
```python
process = subprocess.Popen(
    [self.sim.executable, self.base_name],
    env=self.sim.env,
    stdout=stdout,
    stderr=stderr,
    )

Reply:
> Note This will deadlock when using stdout=PIPE or stderr=PIPE and the child process generates enough
> output to a pipe such that it blocks waiting for the OS pipe buffer to accept more data. Use 
> Popen.communicate() when using pipes to avoid that.